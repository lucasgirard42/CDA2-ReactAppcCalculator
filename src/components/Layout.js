import React from "react";
import Home from "./Home"
import {fetchRandomNumber} from '../actions/calculatorActions'

export default class Layout extends React.Component{
	render(){	

		return(
			<div>
				 <h1>Basic Calculator</h1>
				 <Home fetchRandomNumber={fetchRandomNumber}/>
			</div>
		);
	}
}
