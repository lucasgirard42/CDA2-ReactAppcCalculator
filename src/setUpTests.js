import { configure } from 'enzyme';
import '@testing-library/jest-dom';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import 'jest-enzyme';
import '@testing-library/jest-dom/extend-expect'
configure({ adapter: new Adapter() });