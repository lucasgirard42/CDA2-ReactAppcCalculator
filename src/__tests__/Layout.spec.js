import React from 'react'
import { shallow, mount } from 'enzyme';

import Layout from '../components/Layout';

import '../setUpTests';

//Créer un mock de Home
jest.mock("../components/home", () => {
    return "home-component";
})

    describe('Layout snapshot',()=>{
            
        it('It renders without crashing', () => {
        shallow(<Layout/>)
        // expect(wrapper.length).toEqual(1);
        // expect(wrapper).toMatchSnapshot;
    });

    it('It renders Home', () => {
        const wrapper = mount(<Layout/>);
        //Créer snapshot'
        expect(wrapper.find('home-component'))
        
    });
    it('renders Correctly', () => {
        const wrapper = shallow(<Layout/>);
        //Créer snapshot'
       expect(wrapper).toMatchSnapshot();
        
    });
});