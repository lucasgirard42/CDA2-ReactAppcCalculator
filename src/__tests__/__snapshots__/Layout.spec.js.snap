// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout snapshot It renders without crashing 1`] = `
<div>
  <h1>
    Basic Calculator
  </h1>
  <home-component
    fetchRandomNumber={[Function]}
  />
</div>
`;
